### ISSUE TITLE
Installing Swirl - Ubuntu 14.04, R3.0.2, R Commander 2.0.3 - install.packages("swirl") fails

### ISSUE THREAD

**Issue by Gamp77** [swirldev/swirl#281]
install.packages("swirl")
produces the message in the console:
Warning in install.packages("swirl") :
installation of package 'swirl' had non-zero exit status

In the terminal window, the problem is identified as a dependency, RCurl, which fails to install.
I have tried install.packages("RCurl") but the result is similar, i.e.:

![screenshot](https://cloud.githubusercontent.com/assets/11829942/7017952/9f7f126a-dcf6-11e4-9244-32a86fbee2b5.jpg)

**Comment by seankross** [comment]
Looks like you're having a similar issue as #247. Try that approach and let us know if that works.

**Comment by Gamp77** [comment]
I do apologize for missing that. It was helpful in that it took lading a lot further, but the end result was:
![screenshot from 2015-04-07 08 03 17](https://cloud.githubusercontent.com/assets/11829942/7018466/ba84f01a-dcfc-11e4-9913-870799931cb7.png)

**Comment by seankross** [comment]
Try: `install.packages("swirl", dependencies = TRUE)`

if that doesn't work, you could try (in your shell): `sudo apt-get install r-cran-testthat`

**Comment by Gamp77** [comment]
Great. The second of those two seems to work. Now when I type
> library("swirl")
I get some warning messages about tidying up my workspace before starting, and the last message is
| Type swirl() when you are ready to begin
which I do, and nothing happens.

**Comment by seankross** [comment]
That's bizarre. Have you tried reinstalling swirl?

**Comment by Gamp77** [comment]
I have now. I reinstalled swirl, and got the same messages. Then I cleaned up my workspace before installing, and I got fewer messages than before. But the remaining message appears to imply that I have installed swirl in the wrong place. Should I have specified 'lib'? But I do not get an error message when I submit "library("swirl")". The message is as follows:
[3] NOTE: Installing package into '/home/david/R/x86_64-pc-linux-gnu-library/3.0'
(as 'lib' is unspecified)
trying URL 'http://www.stats.bris.ac.uk/R/src/contrib/swirl_2.2.21.tar.gz'
Content type 'application/x-gzip' length 51037 bytes (49 Kb)
opened URL
==================================================
downloaded 49 Kb

By the way, I have tried several different mirror sites, both here in China and in the UK.

**Comment by seankross** [comment]
So what happens when you type `library(swirl)`?

**Comment by Gamp77** [comment]
This is what I put in the Script window, and then I hit Submit:
rm(list=ls())
library(swirl)
swirl()
This is what I get in the Output window:
> rm(list=ls())

> library(swirl)

> swirl()
And this is what I get in the Messages window:
[1] NOTE: R Commander Version 2.0-3: Wed Apr 8 06:05:17 2015
[2] NOTE:
Rcmdr Version 2.0-3
[3] NOTE:
| Hi! Type swirl() when you are ready to begin.

**Comment by seankross** [comment]
If you type:
```
installed.packages()["swirl" == installed.packages()[,1],]
```
into your console, what's the result?

**Comment by Gamp77** [comment]
As below:
> installed.packages()["swirl" == installed.packages()[,1],]
                                              Package                                               LibPath 
                                              "swirl"       "/home/david/R/x86_64-pc-linux-gnu-library/3.0" 
                                              Version                                              Priority 
                                             "2.2.21"                                                    NA 
                                              Depends                                               Imports 
                                       "R (>= 3.0.2)" "stringr, testthat, httr, yaml, RCurl, digest, tools" 
                                            LinkingTo                                              Suggests 
                                                   NA                                                    NA 
                                             Enhances                                               License 
                                                   NA                                               "GPL-3" 
                                      License_is_FOSS                                 License_restricts_use 
                                                   NA                                                    NA 
                                              OS_type                                                MD5sum 
                                                   NA                                                    NA 
                                     NeedsCompilation                                                 Built 
                                                 "no"                                               "3.0.2"

**Comment by Gamp77** [comment]
Actually, it is laid out in two columns on the screen, and it was in my comment, too, before I submitted it.

**Comment by seankross** [comment]
If you enter into your console:
```
library(swirl, lib.loc = "/home/david/R/x86_64-pc-linux-gnu-library/3.0")
swirl()
```
Does an interactive session begin?

**Comment by Gamp77** [comment]
No, I'm afraid not. It still invites me to type swirl() when I am ready to begin, but nothing happens.
Look, if you want to call it a day and do something else, I will quite understand. I can't tell you how grateful I am for your efforts, and how much I have learned in the process, But there has to come a time when you cut your losses.

**Comment by seankross** [comment]
Paging @WilCrofter, maybe you can figure this out?

**Comment by Gamp77** [comment]
Hi Sean,
You were so generous with your time, I thought you deserved an update. In the end I tried to use RStudio on a machine running Windows Vista. (And it worked fine.) In the course of of downloading RStudio, I found that there was a version of RStudio for Ubuntu. (I had thought that RCommander was roughly equivalent to RStudio, and that RStudio ran on Windows and RCommander ran on Ubuntu, because RCommander came packaged with Ubuntu. I was wrong - the syntax checking and error message features of RCommander are very inferior, and in addition swirl does not run, at least in my case. I have now uninstalled RCommander and installed RStudio, and swirl runs beautifully. But I am much improved about downloading packages for R by the whole experience.
